runtime/bcutil/cfreader.c

```diff
@@ -2201,8 +2201,10 @@ checkClass(J9PortLibrary *portLib, J9CfrClassFile* classfile, U_8* segment, U_32
 		goto _errorFound;
 	}
 
-	/* check annotations and interfaces for invalid flags */
-	if ((value & CFR_ACC_INTERFACE) && (value & ~CFR_INTERFACE_CLASS_ACCESS_MASK))
+	/* check annotations and interfaces for invalid flags. If the ACC_INTERFACE flag is set,
+	 * the ACC_FINAL, ACC_SUPER, and ACC_ENUM flags set must not be set.
+	 */
+	if ((value & CFR_ACC_INTERFACE) && (value & CFR_INTERFACE_FORBIDDEN_ACCESS_MASK))
```

Comment:
better to use the macro: `J9_ARE_ALL_BITS_SET(value, CFR_ACC_INTERFACE | CFR_INTERFACE_FORBIDDEN_ACCESS_MASK)`